db/write_batch.cc

```diff
@@ -1112,9 +1112,23 @@ Status WriteBatch::SingleDelete(ColumnFamilyHandle* column_family,
       this, GetColumnFamilyID(column_family), key);
 }
 
-Status WriteBatchInternal::DeleteRange(WriteBatch* b, uint32_t column_family_id,
+Status WriteBatchInternal::DeleteRange(WriteBatch* b,
+                                       uint32_t column_family_id,
+                                       const Slice& begin_key,
+                                       const Slice& end_key) {
+  return WriteBatchInternal::DeleteRange(b, nullptr, column_family_id, begin_key, end_key);
+}
+
+Status WriteBatchInternal::DeleteRange(WriteBatch* b,
+                                       ColumnFamilyHandle* handle,
+                                       uint32_t column_family_id,
                                        const Slice& begin_key,
                                        const Slice& end_key) {
+  assert(handle->GetComparator() != nullptr);
+  if (handle->GetComparator()->Compare(begin_key, end_key) > 0) {
```

Comment:
I believe there are also types of tables that do not support delete range.  Can/Should those also be checked here?